apm-sniffer/optional-plugins/logger-plugin/src/main/java/org/apache/skywalking/apm/plugin/logger/LogbackLoggerInterceptor.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.logger;
+
+import ch.qos.logback.classic.Logger;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+
+import java.lang.reflect.Method;
+
+public class LogbackLoggerInterceptor implements InstanceMethodsAroundInterceptor {
+
+    private static final ContextConfig.LoggerConfig CONFIG = ContextConfig.getInstance().getLogbackConfig();
+
+    @Override
+    public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes, MethodInterceptResult result) throws Throwable {
+        Logger logger = (Logger) ((Object) objInst);
+        switch (method.getName()) {
```

Comment:
This `switch` could be easily replaced by a more explicit instrumentation definition, with a flag(enum maybe) as the interceptor constructuor, isn't it?
String match is another performance impact.

Reply:
More interceptors for different methods seem higher performance? Just my thinking.

Reply:
@vcjmhg, I think @wu-sheng wanted the switch gone completely. I think he's in favor of adding more `InstanceMethodsAroundInterceptor` until the switch has been implemented at that level. Correct me if I'm wrong @wu-sheng. If so, I could take a swing at it.

Reply:
Yes, I want `switch` gone.

Reply:
@vcjmhg Have you gotten this?

Reply:
Yes, I got it. And I will try it.

Reply:
@wu-sheng , does each method has its own INTERCEPT_CLASS?
Just like this:
`Log4jLoggerInstrumentation.java`
```java
@Override
public String getMethodsInterceptor() {
    if (CONFIG.getLevel() == ContextConfig.LogLevel.FATAL){
        return FATAL_INTERCEPT_CLASS;
    }else if (CONFIG.getLevel() == ContextConfig.LogLevel.ERROR){
        return ERROR_INTERCEPT_CLASS;
    }else if (CONFIG.getLevel() == ContextConfig.LogLevel.WARN){
        return WARN_INTERCEPT_CLASS;
    }else if (...) {
        ...
    }
}
```

Reply:
@wu-sheng , does each method has its own INTERCEPT_CLASS?
Just like this:
`Log4jLoggerInstrumentation.java`
```java
@Override
public InstanceMethodsInterceptPoint[] getInstanceMethodsInterceptPoints() {
    if (CONFIG == null) {
        return null;
    }
    return new InstanceMethodsInterceptPoint[]{
            new InstanceMethodsInterceptPoint() {

                @Override
                public ElementMatcher<MethodDescription> getMethodsMatcher() {
                    return named("fatal")
                }

                @Override
                public String getMethodsInterceptor() {
                    return FATAL_INTERCEPT_CLASS;
                }

                @Override
                public boolean isOverrideArgs() {
                    return false;
                }
            },
            new InstanceMethodsInterceptPoint(){
                 @Override
                public ElementMatcher<MethodDescription> getMethodsMatcher() {
                    return named("error")
                }

                @Override
                public String getMethodsInterceptor() {
                    return ERROR_INTERCEPT_CLASS;
                }

                @Override
                public boolean isOverrideArgs() {
                    return false;
                }
            },
            ...
    };
}
```

Reply:
Yes, I think that would be more efficient, right?

Reply:
Ok, I got it. I will try it.

Reply:
@wu-sheng, I've made the changes you suggested [here](https://github.com/apache/skywalking/pull/5914/commits/3bb27b2cf3963818c74ebc0a8682bd9aef023193).